src/components/Api.js

```diff
@@ -0,0 +1,123 @@
+class Api {
+    constructor({baseUrl, authToken, headers}) {
+        this._baseUrl = baseUrl;
+        this._authToken = authToken;
+        this._headers = headers;
+    }
+
+    getInitialCards() {
+        return fetch("https://around-api.en.tripleten-services.com/v1/cards",)
+    }
+
+        //methods for working with the API 
+        
+        //GET https://around.nomoreparties.co/v1/groupID/cards
+             getCardList(){
+                return fetch(`${this._baseUrl}/cards`, {
+                    headers: {
+                      authorization: this._authToken
+                    }
+                  })
+                    .then(res => {res.ok ? res.json() : Promise.reject(`Error: ${res.status}`)})
+                    .catch((err) => {
+                    console.log(err); // log error to console
+                });
+            }
+
+        // GET https://around.nomoreparties.co/v1/groupID/users/me
+          getUserInfo(){
+        return fetch(`${this._baseUrl}/users/me`, {
+            headers: {
+                authorization: this._authToken
+            }
+        })
+        .then(res => {res.ok ? res.json() : Promise.reject(`Error: ${res.status}`)})
+        .catch((err) => {
+            console.log(err); //log error to console
+        });
+        }
+
+        getAppInfo() {
+          return Promise.all({this:getCardList(), this:getUserInfo()});
+        }
+
+        //POST https://around.nomoreparties.co/v1/groupID/cards
+        addCard({name, link}) {
+           return fetch(`${this._baseUrl}/cards`, {
+                method: "POST",
+                headers: {
+                  authorization: this._authToken,
+                  "Content-Type": "application/json"
+                },
+                body: JSON.stringify({
+                  name,
+                  link
+                })
+              })
+              .then(res => {res.ok ? res.json() : Promise.reject(`Error: ${res.status}`)})
+              .catch((err) => {
+                  console.log(err); //log error to console
+              });
+        }
+
+        //DELETE https://around.nomoreparties.co/v1/groupID/cards/cardID
+        removeCard(cardID) {
+          return fetch(`${this._baseUrl}/cards/${cardID}`, {
+            method: "DELETE",
+            headers: {
+              authorization: this._authToken,
+              "Content-Type": "application/json"
+            }
+          })
+          .then(res => {res.ok ? res.json() : Promise.reject(`Error: ${res.status}`)})
```

Comment:
Needs correction: please reuse `this._handleServerResponse` everywhere for handling the server response